throttle/contract_tx_limiter_middleware_test.go

```diff
@@ -0,0 +1,124 @@
+	udwContract := &udw.UserDeployerWhitelist{}
+	tier := &udwtypes.TierInfo{
+		TierID:     udwtypes.TierID_DEFAULT,
+		Fee:        100,
+		Name:       "Tier1",
+		BlockRange: 10,
+		MaxTx:      2,
+	}
+	tierList := []*udwtypes.TierInfo{}
+	tierList = append(tierList, tier)
+	err := udwContract.Init(contractpb.WrapPluginContext(udwContext), &udwtypes.InitRequest{
+		Owner:    owner.MarshalPB(),
+		TierInfo: tierList,
+	})
+	require.NoError(t, err)
+
+	// Deploy deployerWhitelist and coin contract to deploy a contract for initializing contractToTierMap
+	dwContract := &deployer_whitelist.DeployerWhitelist{}
+	dwAddr := fakeCtx.CreateContract(deployer_whitelist.Contract)
+	dctx := fakeCtx.WithAddress(dwAddr)
+	err = dwContract.Init(contractpb.WrapPluginContext(dctx), &deployer_whitelist.InitRequest{
+		Owner: addr4.MarshalPB(),
+		Deployers: []*dwtypes.Deployer{
+			&dwtypes.Deployer{
+				Address: addr5.MarshalPB(),
+				Flags:   uint32(1),
+			},
+		},
+	})
+	require.Nil(t, err)
+
+	coinContract := &coin.Coin{}
+	coinAddr := fakeCtx.CreateContract(coin.Contract)
+	coinCtx := fakeCtx.WithAddress(coinAddr)
+	err = coinContract.Init(contractpb.WrapPluginContext(coinCtx), &coin.InitRequest{
+		Accounts: []*coin.InitialAccount{
+			{Owner: user, Balance: uint64(300)},
+		},
+	})
+	require.Nil(t, err)
+
+	ret := loom.NewBigUIntFromInt(10)
+	ret.Exp(ret, loom.NewBigUIntFromInt(18), nil)
+	ret.Mul(ret, loom.NewBigUIntFromInt(1000))
+	err = coinContract.Approve(contractpb.WrapPluginContext(coinCtx.WithSender(addr3)), &coin.ApproveRequest{
+		Spender: udwAddr.MarshalPB(),
+		Amount:  &types.BigUInt{Value: *ret},
+	})
+	require.Nil(t, err)
+
+	err = udwContract.AddUserDeployer(contractpb.WrapPluginContext(udwContext.WithSender(addr3)),
+		&udwtypes.WhitelistUserDeployerRequest{
+			DeployerAddr: addr1.MarshalPB(),
+			TierID:       0,
+		})
+	require.Nil(t, err)
+	// record deployment
+	err = udw.RecordEVMContractDeployment(contractpb.WrapPluginContext(udwContext.WithSender(addr3)),
+		addr1, contractAddr)
+	require.Nil(t, err)
+
+	// create middleware
+	state := loomchain.NewStoreState(nil, store.NewMemStore(), abci.Header{Height: 5}, nil, nil)
+	//EVMTxn
+	txSignedEVM := mockSignedTx(t, uint64(1), callId, vm.VMType_EVM, contractAddr)
+	txSignedEVM1 := mockSignedTx(t, uint64(2), callId, vm.VMType_EVM, contractAddr)
+	txSignedEVM2 := mockSignedTx(t, uint64(3), callId, vm.VMType_EVM, contractAddr)
+	cfg := DefaultContractTxLimiterConfig()
+	contractTxLimiterMiddleware := NewContractTxLimiterMiddleware(cfg,
+		func(state loomchain.State) (contractpb.Context, error) {
+			return contractpb.WrapPluginContext(udwContext), nil
+		},
+	)
+
+	ctx := context.WithValue(state.Context(), loomAuth.ContextKeyOrigin, owner)
+	_, err = throttleMiddlewareHandlerCheckTx(contractTxLimiterMiddleware, state, txSignedEVM, ctx)
+	require.NoError(t, err)
+	_, err = throttleMiddlewareHandlerCheckTx(contractTxLimiterMiddleware, state, txSignedEVM1, ctx)
```

Comment:
Since, we care only about whether or not we are reaching next func, you can directly invoke the middleware here. 

something like: 

```
allowed := false

middleware.ProcessTx(...., func(state loomchain.State, txBytes []byte, isCheckTx bool) (res loomchain.TxHandlerResult, err error) {
  allowed = true
  return loomchain.TxHandlerResult{}, nil
})

// Check if allowed should be true or false

```